File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            if target - nums[i] in dict:
                return [i, dict[target-nums[i]]]
            dict[nums[i]] = i
        return [0,0]

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> ans = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(ans.containsKey(target-nums[i])){
                return new int[]{ans.get(target-nums[i]),i};
            }
            ans.put(nums[i], i);
        }
        return new int[]{-1,-1};
    }
}